src/test/java/com/artipie/helm/HelmITCase.java

```diff
@@ -56,172 +34,61 @@
-                new Key.From(HelmITCase.REPO, "index.yaml")
-            ).join(),
-            new IsEqual<>(true)
+    @Test
+    void uploadChartAndCreateIndexYaml() throws Exception {
+        final String url = "http://artipie:8080/my-helm";
+        this.containers.assertExec(
+            "Failed to upload helm archive",
+            new ContainerResultMatcher(new IsEqual<>(0)),
+            "curl", "-X", "PUT", String.format("%s/%s", url, "tomcat"),
+            "--upload-file", String.format("/w/%s", HelmITCase.CHART),
+            "--verbose"
         );
-        con.disconnect();
-    }
-
-    @AfterEach
-    void tearDown() {
-        this.server.stop();
-        this.cntn.close();
-    }
-
-    private void init(final boolean anonymous) throws IOException {
-        this.port = new RandomFreePort().value();
-        this.url = new RepositoryUrl(this.port, "my-helm");
-        this.server = new ArtipieServer(
-            this.tmp, "my-helm", this.config(anonymous).toString(), this.port
+        this.containers.assertExec(
+            "Init failed",
+            new ContainerResultMatcher(new IsEqual<>(0)),
+            "helm", "init", "--stable-repo-url", url, "--client-only"
         );
-        this.server.start();
-        this.cntn = new TestContainer(
-            new GenericContainer<>("alpine/helm:2.16.9")
-                .withCommand("tail", "-f", "/dev/null")
-                .withWorkingDirectory("/home/")
-                .withFileSystemBind(this.tmp.toString(), "/home")
-                .withCreateContainerCmdModifier(
-                    cmd -> cmd.withEntrypoint("/bin/sh")
-                        .withCmd("-c", "while true; do sleep 300; done;")
-                )
+        this.containers.assertExec(
+            "Chart repository was not added",
+            new ContainerResultMatcher(new IsEqual<>(0)),
+            "helm", "repo", "add", "chartrepo", url
         );
-        this.cntn.start(this.port);
-    }
-
-    private RepoConfigYaml config(final boolean anonymous) {
-        final RepoConfigYaml yaml = new RepoConfigYaml("helm")
-            .withFileStorage(this.tmp.resolve("repos"))
-            .withUrl(this.url.string(anonymous))
-            .withPath(this.url.string());
-        if (!anonymous) {
-            yaml.withPermissions(
-                new RepoPerms(ArtipieServer.ALICE.name(), "*")
-            );
-        }
-        return yaml;
-    }
-
-    private String helmRepoAdd(final boolean anonymous, final String chartrepo) throws Exception {
-        final List<String> cmdlst = new ArrayList<>(
-            Arrays.asList("helm", "repo", "add", chartrepo, this.url.string())
+        this.containers.assertExec(
+            "Repo update from chart repository failed",
+            new ContainerResultMatcher(
+                new IsEqual<>(0),
+                new StringContains("Update Complete.")
+            ),
+            "helm", "repo", "update"
```

Comment:
@genryxy as far as I understand this after all these commands some index should be generated on the Artipie side. If so, I'd suggest to check that corresponding file was created, smth like
```
this.containers.assertArtipieContent(
            "Index file was not generated",
            "/var/artipie/data/my-helm/path/to/index",
            new IsNot<>(new IsNull<>())
        );
```

Reply:
@olenagerasimova it implicitly checked by command `helm init --stable-repo-url url`. If there is not index file, this command will fail